pkg/daemon/manager/manager.go

```diff
@@ -109,6 +146,60 @@ func (m *manager) EnsureServiceState(service winsvc.Service, state svc.State) er
 	return winsvc.WaitForState(service, state)
 }
 
+// enumDependentServicesSyscall is a wrapper around enumDependentServicesW.Call with the correct argument casting
+func enumDependentServicesSyscall(hService windows.Handle, dwServiceState uint32, lpServices *byte, cbBufSize uint32,
+	pcbBytesNeeded *uint32, lpServicesReturned *uint32) (uintptr, uintptr, error) {
+	return enumDependentServicesW.Call(uintptr(hService), uintptr(dwServiceState), uintptr(unsafe.Pointer(lpServices)),
+		uintptr(cbBufSize), uintptr(unsafe.Pointer(pcbBytesNeeded)), uintptr(unsafe.Pointer(lpServicesReturned)))
+}
+
+// getDependentServices returns a list of names of all services dependent on the given service
+func (m *manager) getDependentServices(serviceHandle windows.Handle) ([]string, error) {
+	// The documentation for this syscall says it should be ran twice. First to determine the size of the buffer it will
+	// return, and then to actually capture the data with an allocated buffer. In practice, I'm finding it succeeds on
+	// the first run.
+
+	// Borrowing the main steps done here from the golang windows/mgr library's ListServices() function, as the
+	// EnumServicesStatusEx syscall has a very similar way of being called.
+	var buf []byte
+	var bytesNeeded, servicesReturned uint32
+	for {
+		var p *byte
+		if len(buf) > 0 {
+			p = &buf[0]
+		}
+		// Returned error from `Call` will always be non-nil
+		success, _, err := enumDependentServicesSyscall(serviceHandle, windows.SERVICE_STATE_ALL, p, uint32(len(buf)),
+			&bytesNeeded, &servicesReturned)
+		if success != 0 {
+			break
```

Comment:
Is it safe to `break` the loop here? or should `continue` for another attempt?  Consider adding an inline comment here.